trussworks/react-uswds
# docs: Documentation Audit changes

# Summary

It was recently recommended to do an audit of our documentation. This PR includes the changes I recommend.

Additionally, we migrated from the now-deprecated storybook `addon-info` to `addon-docs`, however the parameters api for attaching component descriptions to storybook changed. This PR updates our stories to display the brief descriptions and, more importantly, source links to the original uswds components. https://github.com/trussworks/react-uswds/pull/1008/commits/ed171f924c7455351f2d07a7422317191bd1f4bb For example:

![image](https://user-images.githubusercontent.com/15805554/110690819-58dd6c00-81b2-11eb-9407-4860e2cd8523.png)


Finally, while looking through storybook I was reminded that the documentation around our component props does not populate default values for optional props. This is solved by doing prop restructuring in the functional component signature rather than in the method body. https://github.com/trussworks/react-uswds/pull/1008/commits/27f16c274ff04f83a70b9fd584994afb8b099116 For example:

```jsx
export const GovBanner = ({
    tld = '.gov',
    language = 'english',
    className,
    ...sectionProps
}: GovBannerProps & JSX.IntrinsicElements['section']): React.ReactElement => {
  // Rest of functional component logic/rendering
}
```

Instead of:

```jsx
export const GovBanner = (
  props: GovBannerProps & JSX.IntrinsicElements['section']
): React.ReactElement => {
  const {
    tld = '.gov',
    language = 'english',
    className,
    ...sectionProps
  } = props
  // Rest of functional component logic/rendering
}
```

yielding:

![image](https://user-images.githubusercontent.com/15805554/110549529-2d9a4480-8100-11eb-981d-acf9da2df926.png)

## How To Test

- Mostly copy changes, so proofreading and double checking links would be the best way.
- Unit tests and compilation help ensure confidence that prop destructuring changes didn't break components
- Storybook changes can be seen in the PR storybook deployment.
